server/tserver/src/main/java/org/apache/accumulo/tserver/log/TabletServerLogger.java

```diff
@@ -458,6 +458,26 @@ private void write(final Collection<CommitSession> sessions, boolean mincFinish,
       // the logs haven't changed.
       final int finalCurrent = currentLogId;
       if (!success) {
+
+        if (tserver.getLock() == null || !tserver.getLock().verifyLockAtSource()) {
+          // try to close the log, then Halt the VM
+          testLockAndRun(logIdLock, new TestCallWithWriteLock() {
+
+            @Override
+            boolean test() {
+              return true;
+            }
+
+            @Override
+            void withWriteLock() throws IOException {
+              close();
```

Comment:
This has the potential to block and prevent the thread from calling Halt.  Wondering if this should be done, maybe just skip it and halt.  After we know the lock is not held the sooner we lock the better.